### PR TITLE
8351362: Post-process @Strict annotation for testing

### PR DESCRIPTION
**Pull request by liach**
Added a test tool to strictify files and migrated the existing StrictFinalInstanceFieldsTest as an example.

---------
### Progress
- [ ] Change must not contain extraneous whitespace

### Integration blocker
&nbsp;⚠️ Title mismatch between PR and JBS for issue [JDK-8351362](https://bugs.openjdk.org/browse/JDK-8351362)

### Issue
 * [JDK-8351362](https://bugs.openjdk.org/browse/JDK-8351362): [lworld] Post-process @Strict annotation for testing (**Enhancement** - P4) ⚠️ Title mismatch between PR and JBS.

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/valhalla.git pull/1395/head:pull/1395` \
`$ git checkout pull/1395`

Update a local copy of the PR: \
`$ git checkout pull/1395` \
`$ git pull https://git.openjdk.org/valhalla.git pull/1395/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1395`

View PR using the GUI difftool: \
`$ git pr show -t 1395`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/valhalla/pull/1395.diff">https://git.openjdk.org/valhalla/pull/1395.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/valhalla/pull/1395#issuecomment-2718061671)
</details>

### CODE REVIEW

**Comment by liach** [comment]
Looking to integrate this as I have added all functionalities in the RFE (despite the ctor call relocation being extremely primitive)

/integrate

File: test/lib/jdk/test/lib/value/Strict.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
```
[the remark this answers is not in the document]
Comment (liach) [reply test/lib/jdk/test/lib/value/Strict.java:1]: @dansmithcode aims to migrate all usages of the internal strict to this test-specific strict. This is actually the reason behind this PR - annotations in the Java Language must not change program semantics.

[the remark this answers is not in the document]
Comment (RogerRiggs) [reply test/lib/jdk/test/lib/value/Strict.java:1]: That would then apply to NullRestricted also?
That seems like an unnecessary split, since Strict is an internal annotation, it doesn't matter whether it is declared in a test specific library or jdk.internal.value. and it make maintenance harder because they are split.
And not defined or specified anywhere.

[the remark this answers is not in the document]
Comment (dansmithcode) [reply test/lib/jdk/test/lib/value/Strict.java:1]: The idea is that the `@Strict` annotation should live next to the tool that interprets it.

If we decide we need a translation tool in `jdk.internal`, then putting `@Strict` there too makes sense. If this is purely a test-writing tool, then `@Strict` belongs in tests.

`@NullRestricted` is interpret by the JVM, so it makes sense that it lives in the JVM's internal API.

**Comment by dansmithcode** [comment]
I would not recommend integrating until https://github.com/openjdk/valhalla/pull/1389 is done.

At that point, there will be a large number of tests to experiment with, and after seeing if/how conveniently they can be updated to adopt this tool, we can then have a conversation with the rest of the team about whether this is a good solution going forward, or whether there are still concerns about this not being built into javac.

File: test/lib/jdk/test/lib/value/Strict.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (RogerRiggs) [reply test/lib/jdk/test/lib/value/Strict.java:1]: It has been said that in tests every NullRestricted annotated element should also have \@Strict.
Since \@Strict is there only for test generation; would it be simpler to have the tool look for NullRestricted and do the right thing; avoiding both excessive editing of tests and parallel but dependent annotations.

[the remark this answers is not in the document]
Comment (dansmithcode) [reply test/lib/jdk/test/lib/value/Strict.java:1]: Possibly, yes. The trade-off is then it's harder or confusing to write tests of validation behavior—when you explicitly want a class file that has `@NullRestricted` but not ACC_STRICT. Maybe that's a corner case we can deal with though. It also obscures the dependency on the translation tool (importing the test library is a helpful signal that you intend to use the translation tool). But, sure, that's a possible enhancement.

**Comment by liach** [comment]
/open

This does seems a bit weird in the context of lworld, where many structures already make use of strict - I think this would be more convincing if we can port strict onto simply master, or add a flag on lworld to run with just mainline + strict, without any value object.
